### ISSUE TITLE
heapster issues in v0.12.0

### ISSUE THREAD

**Issue by jackfrancis** [Azure/acs-engine#2083]
**Is this an ISSUE or FEATURE REQUEST?** (choose one): ISSUE

---

**What version of acs-engine?**: v0.12.0

---

**Comment by jackfrancis** [comment]
Example heapster pod logs from a newly deployed cluster:

```
$ kubectl logs heapster-84bb879958-kmfsq heapster -n kube-system
I0118 15:39:53.875470       1 heapster.go:78] /heapster --source=kubernetes.summary_api:""
I0118 15:39:53.875504       1 heapster.go:79] Heapster version v1.5.0
I0118 15:39:53.875672       1 configs.go:61] Using Kubernetes client with master "https://10.0.0.1:443" and version v1
I0118 15:39:53.875687       1 configs.go:62] Using kubelet port 10255
I0118 15:39:53.876332       1 heapster.go:202] Starting with Metric Sink
I0118 15:39:54.077461       1 heapster.go:112] Starting heapster on port 8082
E0118 15:40:05.076012       1 manager.go:101] Error in scraping containers from kubelet_summary:10.240.0.66:10255: Get http://10.240.0.66:10255/stats/summary/: dial tcp 10.240.0.66:10255: getsockopt: connection refused
E0118 15:40:05.076092       1 manager.go:101] Error in scraping containers from kubelet_summary:10.240.0.5:10255: Get http://10.240.0.5:10255/stats/summary/: dial tcp 10.240.0.5:10255: getsockopt: connection refused
E0118 15:40:05.077037       1 manager.go:101] Error in scraping containers from kubelet_summary:10.240.0.4:10255: Get http://10.240.0.4:10255/stats/summary/: dial tcp 10.240.0.4:10255: getsockopt: connection refused
E0118 15:40:05.078374       1 manager.go:101] Error in scraping containers from kubelet_summary:10.240.0.97:10255: Get http://10.240.0.97:10255/stats/summary/: dial tcp 10.240.0.97:10255: getsockopt: connection refused
W0118 15:40:25.000433       1 manager.go:152] Failed to get all responses in time (got 0/4)
```

**Comment by jackfrancis** [comment]
#2091 fixed this bug, which was subsequently included in the `v0.12.1` patch release here:

- https://github.com/Azure/acs-engine/releases/tag/v0.12.1

To manually fix this if you've built a cluster from v0.12.0 and don't want to re-build, update kubelet config on each node to remove the `--read-only-port=0` runtime config setting. This is set in `/etc/default/kubelet`. `sudo systemctl restart kubelet` on each node host to restart kubelet with the modified configuration.
